index.js

```diff
@@ -27,16 +27,21 @@ function fastifyNext (fastify, options, next) {
           return app.close()
         })
 
-      if (!noServeAssets) {
-        const basePath = app.server.nextConfig.basePath || ''
+      if (noServeAssets) {
+        return next()
+      }
+
+      app.getServer().then(server => {
+        const basePath = server?.nextConfig?.basePath || ''
```

Comment:
Is there any change log for this new behaviour?

Reply:
I did not find any, but I think it was recommended to use `.getServer()` even when `.server` was still working, because `.server` was being set after a promise was resolved, `.getServer()` guarantees you get a server without the possibility of being `undefined`.

Reply:
is it? because you use optional chaining.

Reply:
I don't see any downsides to using optional chaining, especially when working with libraries that may change.

Reply:
You wrote that it guarantees that you have server. But still you use optional chaining. 

https://m.youtube.com/watch?v=VuWzeoIr7J4

Reply:
`.server` is also private, as for the CI failing, that is wierd, all the tests I ran were successful

Reply:
Lol is this a typescript feature? 😅🤣

Reply:
Yes, it is not enforced in javascript. It is different than using '#server'